From ed8ee46a304ba3ae0e5a902edebce42f0c21dc30 Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@redhat.com>
Date: Thu, 22 May 2014 15:38:46 -0400
Subject: [redhat] kernel.spec: Modify spec template to build tmon

Message-id: <1400773126-15294-3-git-send-email-nhorman@redhat.com>
Patchwork-id: 80260
O-Subject: [RHEL 7.1 PATCH v2 2/2] redhat: Modify spec template to build tmon
Bugzilla: 1085101
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/kernel.spec.template b/redhat/kernel.spec.template
index 56b7d9d..44b4a08 100644
--- a/redhat/kernel.spec.template
+++ b/redhat/kernel.spec.template
@@ -308,7 +308,7 @@ BuildRequires: numactl-devel
 %endif
 %endif
 %if %{with_tools}
-BuildRequires: pciutils-devel gettext
+BuildRequires: pciutils-devel gettext ncurses-devel
 %endif
 %if %{with_debuginfo}
 # Fancy new debuginfo generation introduced in Fedora 8/RHEL 6.
@@ -516,7 +516,7 @@ This package provides debug information for package kernel-tools.
 # symlinks because of the trailing nonmatching alternation and
 # the leading .*, because of find-debuginfo.sh's buggy handling
 # of matching the pattern against the symlinks file.
-%{expand:%%global debuginfo_args %{?debuginfo_args} -p '.*%%{_bindir}/centrino-decode(\.debug)?|.*%%{_bindir}/powernow-k8-decode(\.debug)?|.*%%{_bindir}/cpupower(\.debug)?|.*%%{_libdir}/libcpupower.*|.*%%{_libdir}/libcpupower.*|.*%%{_bindir}/turbostat(\.debug)?|.*%%{_bindir}/x86_energy_perf_policy(\.debug)?|XXX' -o kernel-tools-debuginfo.list}
+%{expand:%%global debuginfo_args %{?debuginfo_args} -p '.*%%{_bindir}/centrino-decode(\.debug)?|.*%%{_bindir}/powernow-k8-decode(\.debug)?|.*%%{_bindir}/cpupower(\.debug)?|.*%%{_libdir}/libcpupower.*|.*%%{_libdir}/libcpupower.*|.*%%{_bindir}/turbostat(\.debug)?|.*%%{_bindir}/x86_energy_perf_policy(\.debug)?|.*%%{_bindir}/tmon(\.debug)?|XXX' -o kernel-tools-debuginfo.list}
 
 %endif # with_tools
 
@@ -1043,6 +1043,9 @@ make %{?cross_opts} %{?_smp_mflags} -C tools/power/cpupower CPUFREQ_BENCH=false
    popd
 %endif #turbostat/x86_energy_perf_policy
 %endif
+pushd tools
+make tmon
+popd
 %endif
 
 %if %{with_doc}
@@ -1210,6 +1213,9 @@ install -m644 %{SOURCE2001} %{buildroot}%{_sysconfdir}/sysconfig/cpupower
    make DESTDIR=%{buildroot} install
    popd
 %endif #turbostat/x86_energy_perf_policy
+pushd tools/thermal/tmon
+make INSTALL_ROOT=%{buildroot} install
+popd
 %endif
 
 %endif
@@ -1387,7 +1393,7 @@ fi
 %files -f python-perf-debuginfo.list -n python-perf-debuginfo
 %defattr(-,root,root)
 %endif
-%endif # with_perf
+%endif
 
 %if %{with_tools}
 %files -n kernel-tools -f cpupower.lang
@@ -1408,7 +1414,7 @@ fi
 %{_mandir}/man8/turbostat*
 %endif
 %endif
-
+%{_bindir}/tmon
 %if %{with_debuginfo}
 %files -f kernel-tools-debuginfo.list -n kernel-tools-debuginfo
 %defattr(-,root,root)
-- 
1.7.1